### DISCUSSION TITLE
Passing index for accordion prevents expansion of accordion on clicking accordion button

### DISCUSSION THREAD

**Discussion by subhamsaha-manu** [chakra-ui/chakra-ui#6248] [General]
Hi when I pass index array for the `<Accordion>` then clicking on the accordion button does not expand any particular accordion item. Here is a code sandbox link

https://codesandbox.io/s/spring-tdd-kbw4rc?file=/src/App.js

My actual use case is, I want to click a button on the accordion-item(which shows up on hover) that should fire a graphql mutation and at the end the particular accordion item should be expanded. In my production code I am handling the expansion using a state variable passing it to the `index` prop, just like an example in a discussion (https://github.com/chakra-ui/chakra-ui/discussions/2619#discussioncomment-138749). For simplicity I have just passed a hard coded value in the codesandbox example to show the issue.

**Comment by ThaNico** [comment]
Hello @subhamsaha-manu 
I'm running into the same issue, did you find a way to make it work ?
If I use the defaultIndex, it looks like it won't use the state

Basically what I'm trying to do is similar, I have multiple accordion items but only one opened at a time
When I click a button it's supposed to close the one opened and open the next accordion item

Thanks

**Reply by subhamsaha-manu** [reply]
Hi @ThaNico yes was able to find a hack, not sure if that is the proper solution.

I tried summarizing the solution for you here https://excalidraw.com/#json=VGrNheXoUg_dvuJujEnOw,OEGHBjVF8QzhrYbegbByNQ

Let me know if this helps, also I guess if you want only one Accordion item to be expanded at a time you can avoid passing the `allowMultiple` prop to `<Accordion>`

**Reply by ThaNico** [reply]
Thanks, it helped me a lot !
I just put a screenshot of your solution in case the link dies
Have a good day :)

![image](https://user-images.githubusercontent.com/9906385/191201687-2c72b9a8-5aa6-4ea4-9726-65f377557a33.png)

**Reply by subhamsaha-manu** [reply]
Yes good thinking. Happy that it worked! You too have a good day :)
